subsys/net/ip/net_if.c

```diff
@@ -1807,6 +1811,37 @@ void net_if_call_link_cb(struct net_if *iface, struct net_linkaddr *lladdr,
 	}
 }
 
+static bool need_calc_checksum(struct net_if *iface, bool is_tx)
+{
+#if defined(CONFIG_NET_L2_ETHERNET)
+	enum eth_hw_caps eth_caps;
+
+	if (iface->l2 != &NET_L2_GET_NAME(ETHERNET)) {
+		return true;
+	}
+
+	if (is_tx) {
```

Comment:
You can remove this branch by passing eth_caps directly from the functions below